### ISSUE TITLE
[6.1.0] {{on 'click'}}, <LinkTo> and <menu.linkTo> in dropdown menus dysfunctional

### ISSUE THREAD

**Issue by IBue** [ember-bootstrap/ember-bootstrap#2098]
Output from `ember bootstrap:info`:

```
Npm packages:
ember-bootstrap: ~6.1.0 -> 6.1.0
ember-cli: ~4.12.2 -> 4.12.2
bootstrap: ^5.3.2 -> 5.3.2
bootstrap-sass: n/a
ember-cli-sass: n/a
ember-cli-less: n/a
Bower packages:
bootstrap: n/a
bootstrap-sass: n/a
ember-bootstrap configuration:
bootstrapVersion: 5
importBootstrapCSS: true

```

tested:
* _ember-bootstrap_ === 6.1.0 (6.0.1 OK)
* _ember-source_: 4.8 or 4.12
* _bootstrap_: 4 or 5
* _Chromium_, _Firefox_

**Description:**

In _ember-bootstrap_ 6.1.0 `{{on 'click'}} `modifiers in dropdown menu items don't work at all;
anchors generated by `<LinkTo>` or `<menu.linkTo>` only work as external jumps through `href` but not as in-app route transitions.
Browsers show that there _are_ 'click' event handlers attached to the items in question.

**Examples:**

```handlebars
<BsNavbar  as |navbar|>
   …
   <navbar.nav as |nav|>
      <nav.dropdown as |dropdown|>
        <dropdown.toggle>DropdownToggle</dropdown.toggle>
        <dropdown.menu as |menu|>
          <a {{on 'click' this.onclickAction}} class="dropdown-item">onclickAction</a>
          <menu.linkTo @route="route-a" >Route-A (menu.linkTo)</menu.linkTo>
          <LinkTo @route="route-a" class="dropdown-item">Route-A (LinkTo)</LinkTo>
        </dropdown.menu>
      </nav.dropdown>
    </navbar.nav>
 …
</BsNavbar>
```

```handlebars
<BsDropdown as |dd|>
  <dd.button role="button">
    <span>BsDropdown</span>
  </dd.button>
  <dd.menu as |menu|>
    <a {{on 'click' this.onclickAction}} class="dropdown-item">onclickAction</a>
    <menu.linkTo @route="route-a" >Route-A (menu.linkTo)</menu.linkTo>
    <LinkTo @route="route-a" class="dropdown-item">Route-A (LinkTo)</LinkTo>
  </dd.menu>
</BsDropdown>
```

**Comment by SanderKnauff** [comment]
I can confirm that the `<a>` tag is broken. I will probably look into this tomorrow.
@IBue, LinkTo and `menu.linkTo` seem to work, did you mean that it only works due to the href? 

@simonihmig If I fix this, should we release a 6.1.1 with only this fix or a 6.2.0 that also includes the current Typescript work?

**Comment by jelhan** [comment]
> If I fix this, should we release a 6.1.1 with only this fix or a 6.2.0 that also includes the current Typescript work?

I don't see a risk releasing what has been finished from the TypeScript conversion together with this bug fix. I would just not delay releasing the bug fix until TypeScript conversion is fully done. We should document in changelog that TypeScript conversion is on-going process and types published are not considered as stable (and complete).

**Comment by jelhan** [comment]
> I think we can label all TS migration as `internal`, so it does not show up as a feature in the changelogs. And only announce this as a stable when we are ready?

I would mention them in changelog for marketing purposes. It may motivate people to step in and help getting it over the finish line.

**Comment by simonihmig** [comment]
Yeah, but this is happening with an `internal` label. Isn't that enough? As `enhancement` would suggest it to be a new public feature, that justifies a minor version bump, which is what I wanted to prevent...

**Comment by jelhan** [comment]
> Yeah, but this is happening with an `internal` label. Isn't that enough? As `enhancement` would suggest it to be a new public feature, that justifies a minor version bump, which is what I wanted to prevent...

Got it. I don't mind much to be honest. I think adding a sentence on current TypeScript status on the top of the changelog entry is likely more effective.

**Comment by SanderKnauff** [comment]
@IBue This has been fixed in 6.2.0. 
Could you please try it out and see if the issue is solved for you as well?

**Comment by IBue** [comment]
👍 6.2.0 nails it for _ember-source_ 4.8, 4.12, 5.4
…and there is a new test case for it.
